packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -318,7 +310,12 @@ class IOSDevice extends Device {
     @visibleForTesting Duration? discoveryTimeout,
   }) async {
     String? packageId;
-
+    if (interfaceType == IOSDeviceConnectionInterface.network &&
+        debuggingOptions.debuggingEnabled &&
+        debuggingOptions.disablePortPublication) {
+      _logger.printError('Port publication (publish-port) must be enabled for wireless debugging.');
+      return LaunchResult.failed();
```

Comment:
Messages like this need to be actionable by the user.  What are they supposed to do in this case to fix it?  When we have actionable errors like this we should `throwToolExit`.

Something like:
```suggestion
      throwToolExit('Cannot start app on wirelessly tethered iOS device.  Try running again with the --publish-port flag');
```

Introduced in https://github.com/flutter/flutter/pull/67452 to prevent the pop-up from blocking the app in a CI setting.

Maybe `drive` should default to always publishing, and we update our CI scripts to pass `--no-publish-port` so users manually running tests won't hit this?  wdyt?

Reply:
Updated the error message. 

And thanks for the reminder about `drive`, I misunderstood the first time we talked about it. I tried a different approach, so basically added an override for `disablePortPublication` that checks if the flag wasn't provided as an actual argument and it's a network device, then changes it to be enabled. It's a little bit hacky, but allows `drive` to continue functioning as it was before for all wired devices without needing to update anything. Let me know if you think that's not a good solution.